### PR TITLE
Subscription Documentation

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

This PR adds documentation on subscriptions

Additionally, a couple of minor bugs related to subscriptions have been fixed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-1195.surge.sh

File: docs/modules/ROOT/pages/subscriptions/events/update.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+NOTE: Update events will only be triggered if the properties have changed. An update that leaves the same properties will be ignored.
+
+=== Example
+Considering the following type definitions:
+```graphql
+type Movie {
+    title: String
+    genre: String
+}
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/subscriptions/events/update.adoc:24]:
```suggestion
type Movie {
  title: String
  genre: String
}
```

File: docs/modules/ROOT/pages/subscriptions/events/delete.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* `event`: The event triggering this subscription, in this case it will always be `"DELETE"`.
+* `deleted\[type\]`: The top-level properties of the deleted node, these will be the properties right before the node was deleted. Relationships are not available.
+
+=== Example
+Considering the following type definitions:
+```graphql
+type Movie {
+    title: String
+    genre: String
+}
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/subscriptions/events/delete.adoc:22]:
```suggestion
type Movie {
  title: String
  genre: String
}
```

File: docs/modules/ROOT/pages/subscriptions/events/create.adoc [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+* `event`: The event triggering this subscription, in this case it will always be `"CREATE"`.
+* `created\[type\]`: The properties of the newly created node, only top-level properties, without relationships, are available.
+
+=== Example
+Considering the following type definitions:
+```graphql
+type Movie {
+    title: String
+    genre: String
+}
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/subscriptions/events/create.adoc:23]:
```suggestion
type Movie {
  title: String
  genre: String
}
```

**Review by darrellwarde — DISMISSED** [review]
Broadly looks good, thanks for looking at this!!

One overarching comment is that indentation is pretty mixed between different code blocks - this should really be consistent. Can we sort that before merging?

File: docs/modules/ROOT/pages/subscriptions/getting-started.adoc [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    constructor() {
+        this.events = new EventEmitter();
+    }
+
+    publish(eventMeta) {
+        this.events.emit(eventMeta.event, eventMeta);
+    }
+}
+
+const typeDefs = `
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply docs/modules/ROOT/pages/subscriptions/getting-started.adoc:42]: mmm, do we want to add `gql` that to the getting started guide?
What is the real benefit of adding it? it feels it makes the guide even more vendor-locked

File: docs/modules/ROOT/pages/subscriptions/index.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+= Subscriptions (Beta)
+
+WARNING: Subscriptions are only available as a beta its API may change in the future. It is not recommended to use subscriptions in production environments.
+
+
+GraphQL subscriptions add real-time capabilities to your API. This allows you to listen for changes to the database.
+
+The following subscriptions are available when using `@neo4j/graphql`:
+
+* **xref::subscriptions/events/create.adoc[Create]** - Listen for newly created nodes.
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply docs/modules/ROOT/pages/subscriptions/index.adoc:11]: mmm, trying to describe/define subscriptions with the verb subscribe seems like we are not explaining much :thinking:

[the remark this answers is not in the document]
Comment (darrellwarde) [reply docs/modules/ROOT/pages/subscriptions/index.adoc:11]: I won't die on this hill!

File: docs/modules/ROOT/pages/subscriptions/getting-started.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (darrellwarde) [reply docs/modules/ROOT/pages/subscriptions/getting-started.adoc:42]: I also won't die on this hill!

**Review by darrellwarde — APPROVED** [review]

**Pull request merged**